src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -45,6 +48,64 @@ public TelemetryRecorder(IFileSystemUtils fileSystemUtils, IConfiguration config
         Log = log ?? throw new ArgumentNullException(nameof(log));
     }
 
+    private TelemetryRecorder(IConfiguration configuration, IFileSystemUtils fileSystemUtils)
+    {
+        Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        FileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+    }
+
+    /// <summary>
+    /// Create an instance of TelemtryRecorder without the need for an ILogger.
+    /// </summary>
+    /// <param name="configuration">Current configuration of the tool.</param>
+    /// <param name="fileSystemUtils">Wrapper around file system functions.</param>
+    public static TelemetryRecorder Create(IConfiguration configuration, IFileSystemUtils fileSystemUtils)
+    {
+        return new TelemetryRecorder(configuration, fileSystemUtils);
+    }
+
+    /// <summary>
+    /// Method to log telemetry in conditions when the tool is not able to start execution of workflow.
+    /// </summary>
+    /// <param name="exception">Exception that we want to log.</param>
+    public async Task LogToConsole(Exception exception)
+    {
+        var logger = new LoggerConfiguration()
```

Comment:
Since you don't exactly know when someone can call this method. I'll suggest you to do either of two things?

1) Check if the Log method is null and then just follow your existing logic. If Log is not null then use it instead of creating a new Logger.
2) If you don't want this method to be called hen Log is not null, then throw an exception if Log is not null. 

I suggest the first option but it is your call.

Reply:
Decided 1st option makes the most sense for what we are trying to do here.